frontend/src/studio/components/modals/drum-machine/DrumMachineModal.tsx

```
@@ -0,0 +1,342 @@
+    const uniqueGenres = useMemo(() => ['All', ...new Set(drumSamples.map(s => prettyPrint(s.genre || 'Unknown')))], [drumSamples]);
+    const uniqueTypes = useMemo(() => ['All', ...new Set(drumSamples.map(s => prettyPrint(s.category || 'Unknown')))], [drumSamples]);
+
+    useEffect(() => {
+        let filtered = [...drumSamples];
+
+        // Filter by Search Query (case-insensitive)
+        if (searchQuery) {
+            const lowerCaseQuery = searchQuery.toLowerCase();
+            filtered = filtered.filter(sample => 
+                sample.display_name.toLowerCase().includes(lowerCaseQuery) ||
+                (sample.kit_name && sample.kit_name.toLowerCase().includes(lowerCaseQuery))
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)